devtools/maven/pom.xml

```diff
@@ -15,6 +15,11 @@
     <name>Quarkus - Maven Plugin</name>
 
     <properties>
+        <!-- Not sure exactly why but the Maven plugins need to be compiled with Java 11 -->
+        <maven.compiler.target>11</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.release>11</maven.compiler.release>
```

Comment:
@aloubyansky If I drop these lines and default to the parent Java 17, I end up with the following error when anything tries to use this plugin:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:generate-code (default) on project quarkus-integration-test-grpc-descriptor-sets-parent: Execution default of goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:generate-code failed: Unable to load the mojo 'generate-code' (or one of its required components) from the plugin 'io.quarkus:quarkus-maven-plugin:999-SNAPSHOT': com.google.inject.ProvisionException: Unable to provision, see the following errors:
[ERROR]
[ERROR] 1) No implementation for QuarkusBootstrapProvider was bound.
[ERROR]   while locating GenerateCodeMojo
[ERROR]   at ClassRealm[plugin>io.quarkus:quarkus-maven-plugin:999-SNAPSHOT, parent: ClassLoaders$AppClassLoader@251a69d7]
[ERROR]       \_ installed by: WireModule -> PlexusBindingModule
[ERROR]   while locating Mojo annotated with @Named("io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:generate-code")
[ERROR]
[ERROR] 1 error
[ERROR]
[ERROR] ======================
[ERROR] Full classname legend:
[ERROR] ======================
[ERROR] ClassLoaders$AppClassLoader: "jdk.internal.loader.ClassLoaders$AppClassLoader"
[ERROR] GenerateCodeMojo:            "io.quarkus.maven.GenerateCodeMojo"
[ERROR] Mojo:                        "org.apache.maven.plugin.Mojo"
[ERROR] Named:                       "com.google.inject.name.Named"
[ERROR] PlexusBindingModule:         "org.eclipse.sisu.plexus.PlexusBindingModule"
[ERROR] QuarkusBootstrapProvider:    "io.quarkus.maven.QuarkusBootstrapProvider"
[ERROR] WireModule:                  "org.eclipse.sisu.wire.WireModule"
[ERROR] ========================
[ERROR] End of classname legend:
[ERROR] ========================
[ERROR] 
```